src/store/Subscriptions.ts

```diff
@@ -10,17 +11,52 @@ const runParallel = async (dispatch: Dispatch, actionTypes: Array<string>) => {
   });
 };
 
+type SubscriptionsState = {
+  storageUpdatesSubscription: Nullable<VoidFunction>;
+};
+
+type SubscriptionsContext = {
+  state: SubscriptionsState;
+  dispatch: Dispatch;
+};
+
+function initialState(): SubscriptionsState {
+  return {
+    storageUpdatesSubscription: null,
+  };
+}
+
+const state = initialState();
+
 const actions = {
+  async subscribeToStorageUpdates({ state, dispatch }: SubscriptionsContext) {
+    await dispatch('resetStorageUpdatesSubscription');
+
+    state.storageUpdatesSubscription = debounce(100)(() => {
+      dispatch('syncWithStorage', undefined, { root: true });
```

Comment:
should we wait till this methods will be completed?

Reply:
As this is event handler for localstorage update event, we don't need to wait then this handler, or inner actions will be completed (actions are independent)